algorithms/machine-learning/nearest_neighbor.py

```diff
@@ -0,0 +1,57 @@
+import numpy
+
+# train set for the AND-function
+trainSetAND = {(0,0) : 0, (0,1) :0, (1,0) : 0, (1,1) : 1} 
+
+# train set for light or dark colors
+trainSetLight = {(11, 98, 237) : 'L', (3, 39, 96) : 'D', (242, 226, 12) : 'L', (99, 93, 4) : 'D',
+(232, 62, 32) : 'L', (119, 28, 11) : 'D', (25, 214, 47) : 'L', (89, 136, 247) : 'L',
+(21, 34, 63) : 'D', (237, 99, 120) : 'L', (73, 33, 39) : 'D'}
+
+def distance(x,y):
+    """[summary]
+    HELPER-FUNCTION
+    calculates the (eulidean) distance between vector x and y.
+    
+    We use the numpy libriary
+
+    Arguments:
+        x {[tuple]} -- [vector]
+        y {[tuple]} -- [vector]
+    """
+    assert len(x) == len(y), "The vector must have same length"
+    import math
+    result = ()
+    for i in range(len(x)):
+        result += (x[i] -y[i],)
+    return numpy.linalg.norm(result)
+
+def nearest_neighbor(x, tSet):
+    """[summary]
+    Implements the nearest neighbor algorithm 
+
+    Arguments:
+        x {[tupel]} -- [vector]
+        tSet {[dict]} -- [training set]
+    
+    Returns:
+        [type] -- [result of the AND-function]
+    """
+    assert isinstance(x, tuple) and isinstance(tSet, dict)
+    current_key = ()
+    MAX = 32768 # max value 
```

Comment:
is there a reason for setting this to 32768?

Reply:
@keon I want a high max value.

Reply:
I changed it. Thanks